lua/nvim-treesitter/indent.lua

```diff
@@ -36,13 +36,18 @@ end
 ---@param delimiter string
 ---@return TSNode|nil child
 ---@return boolean|nil is_end
+---@return boolean|nil only_delim_at_start
 local function find_delimiter(bufnr, node, delimiter)
   for child, _ in node:iter_children() do
     if child:type() == delimiter then
       local linenr = child:start()
       local line = vim.api.nvim_buf_get_lines(bufnr, linenr, linenr + 1, false)[1]
       local end_char = { child:end_() }
-      return child, #line == end_char[2]
+      local trimmed_before_delim
+      trimmed_before_delim, _ = line:sub(1, end_char[2] - 1):gsub("[%s%" .. delimiter .. "]*", "")
```

Comment:
I think this has to change, if `delimiter` is `s` for example, `"%s%" .. delimiter` will become `%s%s`

Reply:
Yep, agreed, a proper escape here is more sensible, done.